core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.component.ts

```diff
@@ -401,7 +401,41 @@ export class TranslationModalComponent {
     return this.translateTextService.getActiveIndex() > 0;
   }
 
-  skipActiveTranslation(): void {
+  getTrimmedTranslationContent(): string {
+    const content = Array.isArray(this.activeWrittenTranslation)
+      ? this.activeWrittenTranslation.join('')
+      : this.activeWrittenTranslation || '';
+    return content.trim();
+  }
+
+  confirmUnsavedChanges(isSaveDisabled: boolean): boolean {
+    if (isSaveDisabled) {
+      return true;
+    }
+    const translationContent = this.getTrimmedTranslationContent();
+    if (translationContent !== '') {
+      return window.confirm(
+        'You have unsaved changes. Do you want to discard them?'
+      );
+    }
+    return true;
+  }
+
+  isSaveButtonDisabled(): boolean {
+    return (
+      this.loadingData ||
+      this.uploadingTranslation ||
+      this.activeWrittenTranslation === '' ||
+      this.isSubmitted()
+    );
+  }
+
+  skipActiveTranslation(popup: boolean = true): void {
```

Comment:
I don't know what "popup" means and it doesn't sound like a boolean variable (it seems like an object).

I think the function name could also be clearer (and include jsdoc) to convey better what it is actually doing.